### ISSUE TITLE
Docs: An explanation of how to test changes on your own code on the "Local Development" page

### ISSUE THREAD

**Issue by danvk** [typescript-eslint/typescript-eslint#6750]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

If you're adding a new feature or rule, or fixing a bug, it's a good idea to test it on your own code first before putting up a PR. But it's not really clear to me how to do this, andthe "Local Development" page on the typescript-eslint site doesn't have anything to say about it.

I tried running:

    yarn build
    cd ../myproject
    yarn add -D ../typescript-eslint/packages/eslint-plugin

but got the following error:

```
yarn add v1.22.19
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
[3/5] 🚚  Fetching packages...
[4/5] 🔗  Linking dependencies...
error An unexpected error occurred: "expected hoisted manifest for \"@typescript-eslint/eslint-plugin#@typescript-eslint/type-utils\"".
```

What's the recommended way to test local changes against another project? And can it be documented?

### Affected URL(s)

https://typescript-eslint.io/contributing/local-development

**Comment by bradzacher** [comment]
As maintainers we don't generally test things on other repos like that, but I would guess that using [`yarn link`](https://classic.yarnpkg.com/lang/en/docs/cli/link/) would probably work?

Another option is just the ol-copy+pasta - `yarn build && cp packages/eslint-plugin/dist ../my-project/node_modules/@typescript-eslint/eslint-plugin` (or whatever the CP command is supposed to be I forget heh).

Some people also go the route of developing locally in their project and then moving the rule to our codebase.

**Comment by bradzacher** [comment]
We now have docs on this: https://typescript-eslint.io/contributing/local-development/local-linking
